client/reducers/articlesReducer.js

```diff
@@ -11,29 +12,26 @@ export const initialState = {
 export function articlesReducer (state = initialState, action) {
   switch (action.type) {
     case actions.EDITED_ARTICLES_RECEIVED: {
-      return {
-        ...state,
+      return u({
         articlesInSession: action.payload
-      }
+      }, state)
     }
     case editActions.START_EDITING_ARTICLE: {
-      const articlesInSession = clone(state.articlesInSession)
+      const articlesInSession = cloneDeep(state.articlesInSession)
       const session = action.payload
       articlesInSession[session.article] = session
 
-      return {
-        ...state,
+      return u({
         articlesInSession
-      }
+      }, state)
```

Comment:
I'm not familiar with the particulars of this data structure (e.g., possible backbone data, etc), but the use of `cloneDeep` in this file should be unnecessary with updeep / redux as each cycle should return [a pseudo-immutable (frozen) copy](https://github.com/substantial/updeep#about). Thinking something like

```javascript
u({
  articlesInSession: {
    [session.article]: session
  }
}, state)
```
would work similarly well.

Updeep offers a number of functional ways to update data, but simple object updates are typically sufficient. However, [`u.constant`](https://github.com/substantial/updeep#uconstantobject) is really handy if you want to replace something outright.